android/src/main/java/io/ably/flutter/plugin/AblyEventStreamHandler.java

```diff
@@ -0,0 +1,107 @@
+package io.ably.flutter.plugin;
+
+import android.os.Handler;
+import android.os.Looper;
+
+import io.ably.lib.realtime.ChannelStateListener;
+import io.ably.lib.realtime.ConnectionStateListener;
+import io.flutter.plugin.common.EventChannel;
+
+
+public class AblyEventStreamHandler implements EventChannel.StreamHandler {
+    private final AblyMethodCallHandler methodCallHandler;
+
+    AblyEventStreamHandler(AblyMethodCallHandler methodCallHandler){
+        this.methodCallHandler = methodCallHandler;
+    }
+
+    /**
+     * Refer to the comments on AblyMethodCallHandler.MethodResultWrapper
```

Comment:
I've just had a look at those comments and couldn't work out why this is required. Is there something missing either there or here or have the comments gone out of sync?

Reply:
@QuintinWillison 

The comment in `AblyMethodCallHandler.MethodResultWrapper` is basically pointing to this issue https://github.com/flutter/flutter/issues/34993#issue-459900986